examples/cargo-example-derive.rs

```diff
@@ -4,6 +4,7 @@ use clap::Parser;
 #[command(name = "cargo")]
 #[command(bin_name = "cargo")]
 enum Cargo {
+    #[command(name = "example-derive")]
```

Comment:
I feel like overriding the auto-generated name with the same value doesn't help clarify but can also confuse.

I also feel like this example is the wrong place to be teaching how to override a subcommand name.  This is more focused on how to integrate with cargo but this change is more about how to use the attributes.

Reply:
At the end of the day I’m looking for something that nudges people toward getting a working cargo sub command. I want to save others the time I spent staring at my code confused.

Part of that process is understanding how to get the command name they want. It’s unclear (to me) what comes from convention versus what comes from configuration.

Having the base Cli given an explicit name of “cargo” hints to me that there are ways to override defaults. I just couldn’t figure out where to do that.

If you don’t like the renaming how about a comment hinting where the other value is coming from.

Maybe something like:

    // Subcommand “example-derive” generated from variant name

Reply:
Updated with a comment and renamed the args struct to better differentiate where the subcommand name comes from.